R/utils.R

```diff
@@ -113,50 +127,122 @@ is_unknown_os <- function() {
   get_os_type() == "unknown"
 }
 
-#' Create a character object that signals `no_options`
+#' An upgraded version of `Sys.which()` that returns a better Windows path
 #'
+#' @param name A single-length character vector with the executable name.
 #' @noRd
-no_options <- function() {
-  no_opts <- "no_options"
-  class(no_opts) <- "no_options"
-  no_opts
+sys_which <- function(name) {
+
+  # Only accept a vector of length 1
+  stopifnot(length(name) == 1)
+
+  # Get the binary path on Windows
+  if (is_windows_os()) {
+
+    suppressWarnings(
+      system_call <- system(sprintf("where %s", name), intern = TRUE)
+    )
+
+    path_status <- (system_call %>% attributes())$status
+    binary_path <- system_call[1]
+
+    if (!is.null(path_status) && path_status == 0) {
+
+      pathname <-
+        binary_path %>%
+        tidy_gsub("\\\\", "/")
+
+      return(stats::setNames(pathname, name))
+    }
+  }
+
+  Sys.which(name) %>% tidy_gsub("\\\\", "/")
 }
 
-#' Create a character object that signals `no_arg`
+#' Find a binary on the system path or working directory
 #'
+#' @param bin_name The name of the binary to search for.
 #' @noRd
-no_arg <- function() {
-  no_arg_ <- "no_arg"
-  class(no_arg_) <- "no_arg"
-  no_arg_
+find_binary <- function(bin_name) {
+
+  # Find binary on path with `sys_which()`
+  which_result <- sys_which(name = bin_name) %>% unname()
+
+  if (which_result != "") {
+    return(which_result)
+  }
+
+  # Attempt to locate the binary in the working directory
```

Comment:
If `sys_which` already does this for Windows, then get rid of it; otherwise, only do this check on Windows